runners/spark/src/main/java/org/apache/beam/runners/spark/SparkPipelineResult.java

```diff
@@ -149,6 +150,9 @@ public MetricResults metrics() {
     @Override
     protected void stop() {
       SparkContextFactory.stopSparkContext(javaSparkContext);
+      if (Objects.equals(state, State.RUNNING)) {
```

Comment:
I'll leave it that way until the `PipelineResult` API officially stops using `null` as a valid `State`